Resources/public/js/views/serverside/ez-roleserversideview.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-roleserversideview', function (Y) {
+    "use strict";
+    /**
+     * Provides the role server side view
+     *
+     * @module ez-roleserversideview
+     */
+    Y.namespace('eZ');
+
+    var events = {
+            '.ez-role-assign-button': {
+                'tap': '_pickSubtree'
+            },
+        };
+
+    /**
+     * The role server side view. It adds the handling of the role assign
+     * button.
+     *
+     * @namespace eZ
+     * @class RoleServerSideView
+     * @constructor
+     * @extends eZ.ServerSideView
+     */
+    Y.eZ.RoleServerSideView = Y.Base.create('roleServerSideView', Y.eZ.ServerSideView, [], {
+        initializer: function () {
+            this.events = Y.merge(this.events, events);
+        },
+
+        /**
+         * tap event handler on the role assign buttons. It launches the
+         * universal discovery widget so that the user can pick some contents.
+         *
+         * @method _pickSubtree
+         * @protected
+         * @param {EventFacade} e
+         */
+        _pickSubtree: function (e) {
+            var button = e.target,
+                unsetLoading = Y.bind(this._uiUnsetAssignRoleLoading, this, button);
+
+            e.preventDefault();
+            this._uiSetAssignRoleLoading(button);
+            this.fire('contentDiscover', {
+                config: {
+                    title: button.getAttribute('data-universaldiscovery-title'),
+                    cancelDiscoverHandler: unsetLoading,
+                    multiple: true,
```

Comment:
`config` should contain here the `contentDiscoveredHandler` function, let's name it `_contentSelected`. This function will be called after selecting the content from UDW, and selected content will be passed to this function. So this function should trigger event in viewService responsible for assigning the role, see https://github.com/ezsystems/PlatformUIBundle/pull/344#discussion_r40299143